### ISSUE TITLE
hide text alternative for generated content in direction links

### ISSUE THREAD

**Issue by electronicwoft** [govau/design-system-components#702]
the text alternative provided by generated content for the direction links is:
1. inconsistent across browsers (i.e., IE11 doesn't announce generated text etc.)
2. it's annoying if you use a screen reader to hear 'leftward facing arrow back', for example  - it wouldn't be acceptible if this text was onscreen and neither it should be for screen reader users.

perhaps provide guidance to consumers that this class needs to be on a <span> with aria-hidden="true" so the TA for the generated cotnent is not exposed to screen reader users?

**Comment by sukhrajghuman** [comment]
hey @electronicwoft welcome!

This is a really valid point. Thank you so much for bringing it up. I agree with you 100%. I don't think the direction arrows being read out provide much value for the screen reader user.

Also I feel the text in the direction links could be more clear. Right now for example we have `top`, but that could possibly be `top of page` ? What do you think? I'll give with the accessibility team too.

**Comment by electronicwoft** [comment]
thanks for the thumbs up and the welcome ... I wonder does anyone actually use those in-page navigation links? I don't because I use a screen reader and there are much handier ways of getting around, but do other people?

**Comment by sukhrajghuman** [comment]
@electronicwoft I’ve had a chat with the a11y team and they concur that the direction arrows should not be read out. It’s more of a visual cue. Right now the arrow is being added via CSS so we may need to restructure how we deliver this component.

**Comment by sukhrajghuman** [comment]
>I wonder does anyone actually use those in-page navigation links

we have some guidance around some [use cases](https://designsystem.gov.au/components/direction-links/#default) for this component. I think good use case is for users to jump between sections on a page. For example a long content page with links to top of the page.

>much handier ways of getting around

You were talking about landmarks I'm guessing? Screenreader users can navigate through landmarks, but I think it’s useful for non screen reader users.
That being said, a lot of screen reader users [don’t use landmarks](https://webaim.org/projects/screenreadersurvey7/) so this could be useful for them as well.

**Comment by electronicwoft** [comment]
oh, yes, the WebAIM screen reader survey ... I long for the day when there is  credible user research available on such topics! One of the reasons screen reader users might not be tempted to use in-page navigation hyperlinks is that more often than not they simply don't work ... especially with JAWS.

**Comment by electronicwoft** [comment]
"I feel the text in the direction links could be more clear" ... yes, probably, but this is something that might best be tested with users? I avoid in-page links because I am a screen reader user and for the reasons above, but their ambiguity might be an issue for other people? Are you thinking along the lines of WCAG success criterion 2.4.4?

**Comment by electronicwoft** [comment]
@sukhrajghuman ... any progress with this one?

**Comment by sukhrajghuman** [comment]
@electronicwoft this ticket is in our sprint backlog but we haven't gotten around it yet. You can follow our [open backlog](https://github.com/orgs/govau/projects/7)

**Comment by KiriHoy** [comment]
Hi @electronicwoft, there's a couple of approaches that can be taken with this one,  what do you think about using an ```aria-label``` on the anchor tag for the direction link so that the generated arrow icon is not picked up by a screen reader?

An example of the markup for this would be as below:
```<a aria-label="Back" class="au-direction-link au-direction-link--left" href="#">Back</a>```

**Comment by electronicwoft** [comment]
It already has an accessible name provided by the text node of the hyperlink and this is also the label. I have already seen implementations in government in which additional text is added such as 'back to home' or 'return to previous screen' on the back direction arrow and, unless there's going to be some JavaScript to synchronise the accessible name and the label, then there's a distinct possibility that the accessible name and the label will become mismatched. I’d recommend (as does font-awesome) using aria-hidden on a prepended span element for before pseudo elements and an appended span element for after pseudo elements ...
